src/hotspot/share/prims/jvmtiRawMonitor.cpp

```diff
@@ -307,23 +306,35 @@ void JvmtiRawMonitor::simple_notify(Thread* self, bool all) {
   return;
 }
 
-// Any JavaThread will enter here with state _thread_blocked
+// Any JavaThread will enter here with state _thread_blocked, but sometimes not
 void JvmtiRawMonitor::raw_enter(Thread* self) {
   void* contended;
   JavaThread* jt = NULL;
   // don't enter raw monitor if thread is being externally suspended, it will
   // surprise the suspender if a "suspended" thread can still enter monitor
   if (self->is_Java_thread()) {
     jt = self->as_Java_thread();
-    jt->SR_lock()->lock_without_safepoint_check();
-    while (jt->is_external_suspend()) {
-      jt->SR_lock()->unlock();
-      jt->java_suspend_self();
-      jt->SR_lock()->lock_without_safepoint_check();
+    while (true) {
+      // To pause suspends request while in native we must block handshakes.
```

Comment:
// To pause suspend request while in native, we must block handshakes.
s/pauses/pause/